server/server.go

```diff
@@ -42,72 +55,362 @@ func HandleWebsocketConnection(serverHTTPResponse http.ResponseWriter, clientHTT
+	websocketConnection.WriteMessage(websocket.BinaryMessage, messageToClientProtobufBytes)
+	// @TODO HANDLE ERROR HERE?
+}
+
+func deliverMessages(websocketConnection *websocket.Conn, messagesToBeDelivered [][]byte) {
+	// Initialize an empty variable to hold the marshaled protobuf bytes
+	var messageToClientProtobufBytes []byte
+	// Initialize an empty variable to hold a protobuf error in case there is one
+	var messageToClientProtobufErr error
+	// Initialize an empty variable to hold the protobuf message
+	var messageToClientProtobuf backendProtobuf.BackendMessage
+	// Initialize the Request portion of the message
+	messageToClientProtobuf.Request = &backendProtobuf.BackendMessage_Request{}
+	// Initialize the ChatMessage portion of the Request
+	messageToClientProtobuf.Request.Messages = []*backendProtobuf.ChatMessage{}
+	// Iterate over all of the pending messages
+	for _, singleMessageToBeDelivered := range messagesToBeDelivered {
+		// Initialize a ChatMessage protobuf to hold a single message which will be appended to the []*backendProtobuf.ChatMessage{} slice
+		singleMessageToBeDeliveredProtobuf := backendProtobuf.ChatMessage{}
+		// Unmarshal the singleMessageToBeDelivered to the singleMessageToBeDeliveredProtobuf structure
+		protoUnmarshalErr := proto.Unmarshal(singleMessageToBeDelivered, &singleMessageToBeDeliveredProtobuf)
+		if protoUnmarshalErr != nil {
+			// If there is an error
+			messageToClientProtobuf.Error = protoUnmarshalErr.Error()
+		} // if protoUnmarshalErr != nil {
+		// Append the message  to the []*backendProtobuf.ChatMessage{} slice
+		messageToClientProtobuf.Request.Messages = append(messageToClientProtobuf.Request.Messages, &singleMessageToBeDeliveredProtobuf)
+	} // for _, singleMessageToBeDelivered := range messagesToBeDelivered
+	// Marshal the protobuf message structure so that it can be sent to the client over the websocket connection
+	messageToClientProtobufBytes, messageToClientProtobufErr = proto.Marshal(&messageToClientProtobuf)
+	// If we encounter an error while marshaling the protobuf message structure
+	if messageToClientProtobufErr != nil {
+		// Overwrite the current message structure which caused the error with an empty one in hopes that an empty one wont cause an error
+		messageToClientProtobuf = backendProtobuf.BackendMessage{}
+		// Append the error that we encountered when trying to protobuf marshal the original message structure
+		messageToClientProtobuf.Error = messageToClientProtobufErr.Error()
+		// Attempt to marshal the protobuf message structure again which only contains the error we originally encountered so that the client knows there was an error
+		messageToClientProtobufBytes, messageToClientProtobufErr = proto.Marshal(&messageToClientProtobuf)
+	}
+	// Send the mashalled protobuf message structure over the websocket connection
+	websocketConnection.WriteMessage(websocket.BinaryMessage, messageToClientProtobufBytes)
+	// @TODO HANDLE ERROR HERE?
```

Comment:
I would say it's the best to log it here.

Reply:
Yes. I feel if there is an error when trying to ```websocketConnection.WriteMessage``` not much can be done rather than logging it. 

Will implement it.